invokeai/app/invocations/param_easing.py

```diff
@@ -0,0 +1,232 @@
+                                           end=self.end_value,
+                                           duration=base_easing_duration - 1)
+            base_easing_vals = list()
+            for step_index in range(base_easing_duration):
+                easing_val = easing_function.ease(step_index)
+                base_easing_vals.append(easing_val)
+                if print_diagnostics: print("step_index: ", step_index, "easing_val: ", easing_val)
+            if even_num_steps:
+                mirror_easing_vals = list(reversed(base_easing_vals))
+            else:
+                mirror_easing_vals = list(reversed(base_easing_vals[0:-1]))
+            if print_diagnostics:
+                print("base easing vals: ", base_easing_vals)
+                print("mirror easing vals: ", mirror_easing_vals)
+            easing_list = base_easing_vals + mirror_easing_vals
+
+        # FIXME: add alt_mirror option (alternative to default or mirror), or remove entirely
+        # elif self.alt_mirror:  # function mirroring (unintuitive behavior (at least to me))
+        #     # half_ease_duration = round(num_easing_steps - 1 / 2)
+        #     half_ease_duration = round((num_easing_steps - 1) / 2)
+        #     easing_function = easing_class(start=self.start_value,
+        #                                    end=self.end_value,
+        #                                    duration=half_ease_duration,
+        #                                    )
+        #
+        #     mirror_function = easing_class(start=self.end_value,
+        #                                    end=self.start_value,
+        #                                    duration=half_ease_duration,
+        #                                    )
+        #     for step_index in range(num_easing_steps):
+        #         if step_index <= half_ease_duration:
+        #             step_val = easing_function.ease(step_index)
+        #         else:
+        #             step_val = mirror_function.ease(step_index - half_ease_duration)
+        #         easing_list.append(step_val)
+        #         if print_diagnostics: print(step_index, step_val)
+        #
+
+        else:  # no mirroring (default)
+            easing_function = easing_class(start=self.start_value,
+                                           end=self.end_value,
+                                           duration=num_easing_steps - 1)
+            for step_index in range(num_easing_steps):
+                step_val = easing_function.ease(step_index)
+                easing_list.append(step_val)
+                if print_diagnostics: print(step_index, step_val)
+
+        if print_diagnostics:
+            print("prelist size", len(prelist))
+            print("easing_list size", len(easing_list))
+            print("postlist size", len(postlist))
+
+        param_list = prelist + easing_list + postlist
+
+        if self.show_easing_plot:
```

Comment:
I have to make a note that if a user has a backend hosted on a different system, then these plots will not show. Maybe the best way to handle this would to be squash the plot into an image and return that image.

That way we'd be able to utilize it on the frontend irrespective of where the backend is hosted.

But for the time being, this serves the purpose for local usage.

Reply:
Right now if we save a plot image, would that mean it would end up in the image gallery? 
I like the idea someone brought up a few days ago about in-node rendering of the easing plot, eventually.

Reply:
It's pretty straightforward to display a plot in javascript - there are many libraries to handle this. Suggest leaving the node as is instead of outputting to an image. We'll remove this boolean field once the visualisation is implemented.

Reply:
Sounds good!